basedirs/tsv.go

```diff
@@ -0,0 +1,134 @@
+package basedirs
+
+import (
+	"bufio"
+	"bytes"
+	"errors"
+	"io"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/wtsi-ssg/wrstat/v4/internal/split"
+)
+
+type ConfigAttrs struct {
+	Prefix  []string
+	Splits  uint64
+	MinDirs uint64
+}
+
+type Config []ConfigAttrs
+
+var (
+	ErrBadTSV = errors.New("bad TSV")
+
+	slashByte   = []byte{'/'}  //nolint:gochecknoglobals
+	newLineByte = []byte{'\n'} //nolint:gochecknoglobals
+	tabByte     = []byte{'\t'} //nolint:gochecknoglobals
+)
+
+const (
+	numColumns = 3
+	noMatch    = -1
+
+	DefaultSplits  = 1
+	defaultMinDirs = 2
+)
+
+func ParseConfig(r io.Reader) (Config, error) {
```

Comment:
Document. Change Config to not support * as wildcard, and therefore we can just do simple prefix matches